controller_manager/src/controller_manager.cpp

```diff
@@ -2501,20 +2501,23 @@ bool ControllerManager::controller_sorting(
       // If there is no common parent, then they belong to 2 different sets
       auto following_ctrls_b = get_following_controller_names(ctrl_b.info.name, controllers);
       if (following_ctrls_b.empty()) return true;
-      auto find_first_element = [&](const auto & controllers_list)
+      auto find_first_element = [&](const auto & controllers_list) -> long int
       {
         auto it = std::find_if(
           controllers.begin(), controllers.end(),
           std::bind(controller_name_compare, std::placeholders::_1, controllers_list.back()));
         if (it != controllers.end())
         {
-          int dist = std::distance(controllers.begin(), it);
-          return dist;
+          return std::distance(controllers.begin(), it);
         }
+        return 0;
```

Comment:
@destogl Thanks for fixing the warnings

Regarding returning 0, I don't think there might be a change in behavior as the controllers that are parsed to this lambda are always a subset of the main controller list. I think for this reason, I've never encountered an issue of the controller not being found inside the lambda. I think returning zero is also good as a default behavior.

What do you think?

Reply:
That is also how I have understood the code. So basically giving 0 if no controller is found in the list then in the following if we are checking `0 < 0` which is fine to return `false`.

I just wanted to check with you that I didn't make a logical mistake.

Reply:
Yeah. looks good to me

Reply:
OK, please appove it then :D